BrightFutures/AsyncType.swift

```diff
@@ -75,8 +75,10 @@ public extension AsyncType {
     /// queue.
     public func delay(queue: Queue, interval: NSTimeInterval) -> Self {
         return Self { complete in
-            queue.after(.In(interval)) {
-                self.onComplete(ImmediateExecutionContext, callback: complete)
+            onComplete(queue.context) { result in
```

Comment:
This can be `onComplete(ImmediateExecutionContext) { result in`. We're not doing any thread-sensitive work inside the closure and the immediate context saves us one call to `dispatch_async`.